docs/user-guide/extending.rst

```diff
@@ -0,0 +1,14 @@
+
+Extending Zarr
+==============
+
+Zarr-Python 3 was designed to be extensible. This means that you can extend
+the library by writing custom classes and plugins. Currently, Zarr can be extended
+in the following ways:
+
+1. Writing custom stores
+2. Writing custom codecs
```

Comment:
@normanrz - could you suggest a short bit of text on how the codec system is meant to work.